src/db/car/forest.rs

```diff
@@ -242,12 +242,10 @@ where
 
 fn decode_zstd_single_frame<ReaderT: Read>(reader: ReaderT) -> io::Result<BytesMut> {
     let mut zstd_frame = vec![];
-
     zstd::Decoder::new(reader)?
         .single_frame()
         .read_to_end(&mut zstd_frame)?;
-    // This unnecessarily copies the zstd frame. :(
-    Ok(BytesMut::from(zstd_frame.as_slice()))
+    Ok(zstd_frame.into_iter().collect())
```

Comment:
Won't `collect` do the copy on its own?

Reply:
I think it should transfer ownership and avoid copy or clone. In below sample, `A` does not derive `Copy` or `Clone`
 ```rust
#[derive(Debug)]
struct A;

fn main() {
    let a = vec![A, A, A];
    let b: Vec<_> = a.into_iter().collect();
    println!("{b:?}");
}
```

Reply:
Also checkout https://github.com/ChainSafe/forest/pull/4233#issuecomment-2061553746

Reply:
I need to go back to Rust for Dummies.